src/utils/parser_server.rb

```diff
@@ -16,32 +12,61 @@
 trap(:INT) { quit = true }
 trap(:TERM) { quit = true }
 
-server = TCPServer.new(ARGV.first || 22_021)
+sockfile = ARGV.first
+server = UNIXServer.new(sockfile)
+
+at_exit do
+  server.close
+  File.unlink(sockfile)
+end
+
+MAX_LEN = 10 * 1024 * 1024
```

Comment:
The biggest file size you can send is limited by the `maxBuffer` option on the node process, so let's just go with that (15M) and avoid the `read_message` loop below here